### ISSUE TITLE
Error in user changeset

### ISSUE THREAD

**Issue by jonathanraes** [danschultzer/pow_assent#44]
Hey there,
The documentation provides the following changeset

```
def user_identity_changeset(user_or_changeset, user_identity, attrs, user_id_attrs) do
    user_or_changeset
    |> Ecto.Changeset.cast(attrs, [:custom_field_1, :customer_field_2])
    |> pow_assent_user_identity_changeset(user, user_identity, attrs, user_id_attrs)
end
```

I had 2 problems with this piece, first I think the parameter `user` to `pow_assent_user_identity_changeset` should not be there (it generated an error while compiling). 
Second, this only worked for me when I switched the 2 calls around, the `pow_assent_user_identity_changeset` removed my custom fields for some reason. This is the resulting method:

```
def user_identity_changeset(user_or_changeset, user_identity, attrs, user_id_attrs) do
    user_or_changeset
    |> pow_assent_user_identity_changeset(user_identity, attrs, user_id_attrs)
    |> Ecto.Changeset.cast(attrs, [:custom_field_1, :customer_field_2])
end
```

**Comment by danschultzer** [comment]
Thanks. I've updated the docs. The order shouldn't matter though, I've just tested it, and added a test where the cast happens before the `pow_assent_user_identity_changeset/4` call, and it works.

**Comment by jonathanraes** [comment]
No problem. Strange, Ive just tested it again and it definitely matters for me. Im using it to add the `:email` and `:name` field, when I do the cast first and inspect the resulting values, the fields are there after the cast and have dissapeared after the `pow_assent_user_identity_changeset` call.

**Comment by danschultzer** [comment]
Can you show me how your `user_identity_changset` and `changeset` methods looks like in your user schema module?

Also, by default `:email` would be handled by the `pow_assent_user_identity_changeset` method too.

**Comment by jonathanraes** [comment]
These are my changesets now :
```
def changeset(user_or_changeset, attrs) do
    user_or_changeset
    |> pow_changeset(attrs)
    |> pow_extension_changeset(attrs)
    |> Ecto.Changeset.cast(attrs, [:email, :name])
    |> Ecto.Changeset.validate_required([:email])
  end

  def user_identity_changeset(user_or_changeset, user_identity, attrs, user_id_attrs) do
    user_or_changeset
    |> pow_assent_user_identity_changeset(user_identity, attrs, user_id_attrs)
    |> Ecto.Changeset.cast(attrs, [:email, :name])
  end
```

Even when i remove the `user_identity_changeset` method entirely I dont get the email field set in my database.

**Comment by danschultzer** [comment]
Is it only the e-mail field that doesn't get set, or is it also the name field?

If it's only the e-mail field then it's because `pow_assent_user_identity_changeset/4` will set that for you (since it's the default user id).

**Comment by danschultzer** [comment]
Just FYI, I made the following test, and it sets the email and name when casting before the `pow_assent_user_identity_changeset/4` :

```elixir
defmodule TestUser do
  use Ecto.Schema
  use Pow.Ecto.Schema
  use PowAssent.Ecto.Schema

  schema "users" do
    field :name, :string

    pow_user_fields()

    timestamps()
  end

  def user_identity_changeset(user_or_changeset, user_identity, attrs, user_id_attrs) do
    user_or_changeset
    |> Ecto.Changeset.cast(attrs, [:name, :email])
    |> Ecto.Changeset.validate_required([:name])
    |> pow_assent_user_identity_changeset(user_identity, attrs, user_id_attrs)
  end
end

TestUser.user_identity_changeset(%TestUser{}, %{provider: "test_provider", uid: "1"}, %{email: "test@example.com", name: "John Doe"}, nil) |> IO.inspect()

#Ecto.Changeset<
  action: nil,
  changes: %{
    email: "test@example.com",
    name: "John Doe",
    user_identities: [
      #Ecto.Changeset<
        action: :insert,
        changes: %{provider: "test_provider", uid: "1"},
        errors: [],
        data: #TestUser.UserIdentities.UserIdentity<>,
        valid?: true
      >
    ]
  },
  errors: [],
  data: #TestUser<>,
  valid?: true
>

```

**Comment by jonathanraes** [comment]
None of them get set, the email was immediately obvious though because my database enforces non-null for email.

The log shows only 3 values are getting set:
``` 
[debug] QUERY ERROR db=9.9ms
INSERT INTO "users" ("email_confirmed_at","inserted_at","updated_at") VALUES ($1,$2,$3) RETURNING "id" [#DateTime<2019-03-21 18:24:23Z>, ~N[2019-03-21 18:24:23], ~N[2019-03-21 18:24:23]]
```

I have no idea what could be causing this, but at least its working correctly with my current setup.

**Comment by danschultzer** [comment]
Finally figured it out (with same issue in #49). It's when PowEmailConfirmation has been enabled. I've released 2.0.2 that resolves it!

**Comment by jonathanraes** [comment]
Works great now thanks
